File: src/main/java/dev/losterixx/sCore/features/autobroadcaster/BroadcastManager.java
package dev.losterixx.sCore.features.autobroadcaster;

import dev.dejvokep.boostedyaml.YamlDocument;
import dev.losterixx.sCore.Main;
import dev.losterixx.sCore.utils.ConfigManager;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Bukkit;
import org.bukkit.Sound;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class BroadcastManager {

    private static MiniMessage mm = Main.mm;
    private static Main main = Main.getInstance();
    private static ConfigManager configManager = main.getConfigManager();
    private static YamlDocument config = configManager.getConfig("config");
    private static Component prefix = mm.deserialize(config.getString("prefix"));

    public static void broadcast(String messageId) {
        if (!config.contains("autoBroadcaster.messages." + messageId)) {
            main.getLogger().warning("Message with ID " + messageId + " could not be found!");
            return;
        }

        List<String> messagesToSent = config.getStringList("autoBroadcaster.messages." + messageId);

        if (messagesToSent.isEmpty()) {
            main.getLogger().warning("Message with ID " + messageId + " is empty!");
            return;
        }

        for (String message : messagesToSent) {
            if (message.replaceAll(" ", "").isEmpty()) {
                message = "<gray> ";
            }

            for (Player player : Bukkit.getOnlinePlayers()) {
                player.sendMessage(prefix.append(mm.deserialize(message)));
            }

            Bukkit.getConsoleSender().sendMessage(prefix.append(mm.deserialize(message)));
        }

        if (config.getBoolean("autoBroadcaster.sound.enabled")) {
            Sound sound = Sound.valueOf(config.getString("autoBroadcaster.sound.name"));
            double volume = config.getDouble("autoBroadcaster.sound.volume");
            double pitch = config.getDouble("autoBroadcaster.sound.pitch");

            main.getServer().getOnlinePlayers().forEach(player -> {
                player.playSound(player, sound, (float) volume, (float) pitch);
            });
        }
    }

    public void startBroadcasting() {
        int interval = config.getInt("autoBroadcaster.interval");

        if (interval <= 0)
            return;

        short type = config.getShort("autoBroadcaster.type");

        if (!config.getBoolean("autoBroadcaster.enabled"))
            return;

        new BukkitRunnable() {
            private int index = 0;

            @Override
            public void run() {
                if (type == 1) { // SEQUENTIAL
                    List<String> keys = new ArrayList<>();
                    for (Object key : config.getSection("autoBroadcaster.messages").getKeys()) {
                        keys.add((String) key);
                    }
                    if (keys.isEmpty()) return;

                    String messageId = keys.get(index);
                    broadcast(messageId);

                    index = (index + 1) % keys.size();
                } else if (type == 2) { // RANDOM
                    List<String> keys = new ArrayList<>();
                    for (Object key : config.getSection("autoBroadcaster.messages").getKeys()) {
                        keys.add((String) key);
                    }
                    if (keys.isEmpty()) return;

                    String messageId = keys.get(new Random().nextInt(keys.size()));
                    broadcast(messageId);
                }
            }
        }.runTaskTimer(main, (interval * 20L) / 2, interval * 20L); // interval in seconds
    }

    public static void updateConfigs() {
        config = configManager.getConfig("config");
        prefix = mm.deserialize(config.getString("prefix"));
    }

}

File: src/main/kotlin/dev/losterixx/sCore/features/autobroadcaster/BroadcastManager.kt
package dev.losterixx.sCore.features.autobroadcaster

import dev.losterixx.sCore.Main
import dev.losterixx.sCore.utils.ConfigManager
import org.bukkit.Bukkit
import org.bukkit.Sound
import org.bukkit.SoundCategory
import org.bukkit.scheduler.BukkitRunnable
import kotlin.random.Random


object BroadcastManager {

    private val mm = Main.miniMessage
    private val main = Main.instance
    private fun getConfig() = ConfigManager.getConfig("config")
    private fun getPrefix() = getConfig().getString("prefix") ?: Main.DEFAULT_PREFIX

    fun broadcast(messageId: String?) {
        if (!getConfig().contains("autoBroadcaster.messages.$messageId")) {
            main.logger.warning("Message with ID $messageId could not be found!")
            return
        }

        val messagesToSent = getConfig().getStringList("autoBroadcaster.messages.$messageId")

        if (messagesToSent.isEmpty()) {
            main.logger.warning("Message with ID $messageId is empty!")
            return
        }

        for (msg in messagesToSent) {
            var message = msg
            if (message.replace(" ", "").isEmpty()) {
                message = "<gray> "
            }

            for (player in Bukkit.getOnlinePlayers()) {
                player.sendMessage((mm.deserialize(message
                    .replace("%prefix%", getPrefix()))))
            }

            Bukkit.getConsoleSender().sendMessage(mm.deserialize(message
                .replace("%prefix%", getPrefix())))
        }

        if (getConfig().getBoolean("autoBroadcaster.sound.enabled")) {
            val soundName = getConfig().getString("autoBroadcaster.sound.sound")?.uppercase()?.replace('.', '_') ?: "ENTITY_PLAYER_LEVELUP"
            val sound = try {
                Sound.valueOf(soundName)
            } catch (e: Exception) {
                main.logger.warning("Invalid sound-name for autoBroadcaster: $soundName. Using default sound.")
                Sound.ENTITY_PLAYER_LEVELUP
            }
            val volume = getConfig().getDouble("autoBroadcaster.sound.volume")
            val pitch = getConfig().getDouble("autoBroadcaster.sound.pitch")

            for (player in Bukkit.getOnlinePlayers()) {
                player.playSound(player, sound, SoundCategory.MASTER, volume.toFloat(), pitch.toFloat())
            }
        }
    }

    fun startBroadcasting() {
        if (!getConfig().getBoolean("autoBroadcaster.enabled")) return

        val interval = getConfig().getInt("autoBroadcaster.interval")
        val type = getConfig().getByte("autoBroadcaster.type")

        if (interval <= 0) return

        object : BukkitRunnable() {
            private var index = 0

            override fun run() {
                if (Bukkit.getOnlinePlayers().isEmpty()) return
                if (type.toInt() == 1) { // SEQUENTIAL
                    val keys: MutableList<String?> = ArrayList<String?>()
                    for (key in getConfig().getSection("autoBroadcaster.messages").getKeys()) {
                        keys.add(key as String?)
                    }
                    if (keys.isEmpty()) return

                    val messageId = keys[index]
                    broadcast(messageId)

                    index = (index + 1) % keys.size
                } else if (type.toInt() == 2) { // RANDOM
                    val keys: MutableList<String?> = ArrayList<String?>()
                    for (key in getConfig().getSection("autoBroadcaster.messages").getKeys()) {
                        keys.add(key as String?)
                    }
                    if (keys.isEmpty()) return

                    val messageId = keys[Random.nextInt(keys.size)]
                    broadcast(messageId)
                }
            }
        }.runTaskTimer(main, (interval * 20L) / 2, interval * 20L) // interval in seconds
    }

}